server.go

```diff
@@ -1223,48 +1231,87 @@ func (s *Server) ListenAndServeTLSEmbed(addr string, certData, keyData []byte) e
 // ServeTLS serves HTTPS requests from the given listener.
 //
 // certFile and keyFile are paths to TLS certificate and key files.
+//
+// If the certFile or keyFile has not been provided the server structure,
+// the function will use previously added TLS configuration.
 func (s *Server) ServeTLS(ln net.Listener, certFile, keyFile string) error {
-	lnTLS, err := newTLSListener(ln, certFile, keyFile)
-	if err != nil {
-		return err
+	s.AppendCert(certFile, keyFile)
```

Comment:
You need to check the error returned by this function. If someone now specified and invalid certificate they will get an `No certFile or keyFile has been provided` error which is very confusing and incorrect.